src/ui/popup.js

```diff
@@ -572,8 +572,12 @@ export default class Popup extends Evented {
         }
 
         const offsetedPos = pos.add(offset[anchor]).round();
-        DOM.setTransform(this._container, `${anchorTranslate[anchor]} translate(${offsetedPos.x}px,${offsetedPos.y}px)`);
-        applyAnchorClass(this._container, anchor, 'popup');
+        this._map._domRenderTaskQueue.add(() => {
+            if (this._container && anchor) {
```

Comment:
Nit: `anchor` is always truthy here so no need to check.

Reply:
I agree, but `flow` is not understanding this. Is there any flag I can maybe add to make it understand?

Reply:
Same goes in `marker.js` btw.

Reply:
[See this previous job on CircleCI for reference](https://app.circleci.com/pipelines/github/mapbox/mapbox-gl-js/6204/workflows/288bc121-7bb4-407a-97fd-b8c3d04c9b11/jobs/83235/steps)